ambry-quota/src/main/java/com/github/ambry/quota/AmbryQuotaManager.java

```diff
@@ -128,15 +133,22 @@ public ThrottlingRecommendation charge(RestRequest restRequest, BlobInfo blobInf
       return null;
     }
     ThrottlingRecommendation throttlingRecommendation;
+    List<QuotaRecommendation> quotaRecommendations = new ArrayList<>();
     Timer.Context timer = quotaMetrics.quotaChargeTime.time();
     try {
-      throttlingRecommendation = quotaRecommendationMergePolicy.mergeEnforcementRecommendations(requestQuotaEnforcers.stream()
-          .map(quotaEnforcer -> quotaEnforcer.chargeAndRecommend(restRequest, blobInfo, requestCostMap))
-          .filter(Objects::nonNull)
-          .collect(Collectors.toList()));
+      for (QuotaEnforcer quotaEnforcer : requestQuotaEnforcers) {
+        try {
+          quotaEnforcer.charge(restRequest, blobInfo, requestCostMap);
+          quotaRecommendations.add(quotaEnforcer.recommend(restRequest));
```

Comment:
hmm, we should probably reserve chargeAndRecommend method since it should be an atomic operation in this case.